### PR TITLE
Update react native example.

### PR DESCRIPTION
**Pull request by thruflo**

### CODE REVIEW

**Comment by thruflo** [comment]
@balegas here's the wip on an updated react-native example.

It's a standard `npx react-native init MyApp --template react-native-template-typescript` boilerplate. You need to:

```
yarn
cd ios && pod install && cd ..
```

Plus atm I'm using yalc for the `electric-sql` dependency. In your `./typescript-client` run:

```sh
yalc publish
# or `yarn build && yalc publish --no-script` if you want to skip running tests
```

Then in this repo

```sh
yalc update electric-sql
```

Then you can refresh the simulator to see changes.

File: expo/App.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import { Pressable, StyleSheet, Text, View } from "react-native";
+
+import * as SQLite from "expo-sqlite";
+
+import { electrify } from "electric-sql/dist/drivers/expo-sqlite";
```
Comment (thruflo) [inline expo/App.js:6]: We should use the shorter import paths defined in the exports mapping in package.json. So `import { ... } from 'electric-sql/expo'` in this case.

File: expo/App.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  if (db === undefined || results === undefined) {
+    return null;
+  }
+
+  const addItem = () => {
+    const randomValue = Math.random().toString(16).substr(2);
+
+    db.transaction((tx) => {
+      tx.executeSql("INSERT INTO items VALUES(?)", [randomValue], () => {
+        db.electric.notifier.potentiallyChanged();
```
[the remark this answers is not in the document]
Comment (thruflo) [reply expo/App.js:65]: Note also that the Expo driver should correctly wrap the transaction function as per https://github.com/electric-sql/typescript-client/blob/main/src/drivers/expo-sqlite/database.ts#L101 -- so we should be fine to rely on the automatic notifications here?

File: expo/app.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "expo": {
```
Comment (thruflo) [inline expo/app.json:2]: We have an `electric-sql` Expo account. The docs say:

> Add "owner": "electric-sql" under the "expo" key of app.json

https://docs.expo.dev/versions/latest/config/app/#owner

File: expo/App.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+import { data as migrationsData } from "./migrations";
+import { StatusBar } from "expo-status-bar";
+
+export default Example = () => {
+  const [db, setDb] = useState();
+
+  useEffect(() => {
+    const odb = SQLite.openDatabase("example.db");
+    new Promise((res) =>
```
[the remark this answers is not in the document]
Comment (balegas) [reply expo/App.js:17]: Need the promise. Fixed because we should wait too.

[the remark this answers is not in the document]
Comment (thruflo) [reply expo/App.js:17]: Fwiw, I believe the main pattern is to define an async function inside useEffect and then to call it, as per this sort of thing:

```js
useEffect(() => {
  const fetchData = async () => {
    const data = await fetch('https://yourapi.com')
    setData(data)
  }

  fetchData().catch(console.error)
}, [])
```

[the remark this answers is not in the document]
Comment (balegas) [reply expo/App.js:17]: yes that is what I got. Pushing in a few.

**Pull request merged**